sdlib/d/gc/region.d

```diff
@@ -234,25 +254,43 @@ unittest acquire_release {
 	void* addr0;
 	assert(regionAllocator.acquire(&addr0));
 
+	// check the block counter works
+	assert(regionAllocator.nBlocks == 1);
```

Comment:
The overall checking of the tracking of nBlocks is lacking. It's missing:
1. Checking that the count is maintained when extra blocks are allocated.
2. Checking that the count is not affected by allocation failure (if there is a reasonable way to do this, idk).

Same for remove.

Reply:
For 2, I don't know of a reasonable way. The only way to get a return without allocating is to run out of address space I think.

Reply:
For 1, added more checks in the "extra blocks" unittest.

Reply:
I removed most of the unittests testing the value tracking functionality, because that is now a separate object with its own unittests